<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>741</width>
    <height>519</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <widget class="QDialogButtonBox" name="buttonBox">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>470</y>
     <width>341</width>
     <height>32</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="standardButtons">
    <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
   </property>
  </widget>
  <widget class="QWidget" name="gridLayoutWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>50</y>
     <width>356</width>
     <height>130</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QLabel" name="label">
      <property name="text">
       <string> Select grid height:</string>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QLCDNumber" name="heightLcdNumber">
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <property name="intValue" stdset="0">
       <number>100</number>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QSlider" name="widthSlider">
      <property name="minimum">
       <number>3</number>
      </property>
      <property name="maximum">
       <number>10000</number>
      </property>
      <property name="value">
       <number>100</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>Select grid width:</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>Default layout:</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QSlider" name="heightSlider">
      <property name="minimum">
       <number>3</number>
      </property>
      <property name="maximum">
       <number>10000</number>
      </property>
      <property name="value">
       <number>100</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QComboBox" name="comboBox">
      <item>
       <property name="text">
        <string>Blinker</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Block</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Glider</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Gosper glider gun</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>C heptomino</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Biloaf</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Octagon II</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Copperhead</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Diamond 4 8 12</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QLCDNumber" name="widthLcdNumber">
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="intValue" stdset="0">
       <number>100</number>
      </property>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QPushButton" name="RandpushButton">
      <property name="text">
       <string>randomize</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QLabel" name="label_4">
      <property name="text">
       <string>Random layout:</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Dialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Dialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>heightSlider</sender>
   <signal>sliderReleased()</signal>
   <receiver>Dialog</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>259</x>
     <y>97</y>
    </hint>
    <hint type="destinationlabel">
     <x>370</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widthSlider</sender>
   <signal>sliderReleased()</signal>
   <receiver>Dialog</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>259</x>
     <y>137</y>
    </hint>
    <hint type="destinationlabel">
     <x>370</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widthSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>widthLcdNumber</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>206</x>
     <y>107</y>
    </hint>
    <hint type="destinationlabel">
     <x>307</x>
     <y>107</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>heightSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>heightLcdNumber</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>206</x>
     <y>70</y>
    </hint>
    <hint type="destinationlabel">
     <x>307</x>
     <y>70</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox</sender>
   <signal>activated(int)</signal>
   <receiver>Dialog</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>206</x>
     <y>154</y>
    </hint>
    <hint type="destinationlabel">
     <x>370</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RandpushButton</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>219</x>
     <y>160</y>
    </hint>
    <hint type="destinationlabel">
     <x>370</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
